Sources/Sentry/SentryScope.m

```diff
@@ -0,0 +1,343 @@
+
+/**
+ * Set global user -> thus will be sent with every event
+ */
+@property(atomic, strong) SentryUser *_Nullable userObject;
+
+/**
+ * Set global tags -> these will be sent with every event
+ */
+@property(nonatomic, strong) NSMutableDictionary<NSString *, NSString *> *_Nullable tagDictionary;
+
+/**
+ * Set global extra -> these will be sent with every event
+ */
+@property(nonatomic, strong) NSMutableDictionary<NSString *, id> *_Nullable extraDictionary;
+
+/**
+ * used to add values in event context.
+ */
+@property(nonatomic, strong) NSMutableDictionary<NSString *, NSDictionary<NSString *, id>*> *_Nullable contextDictionary;
+
+/**
+ * Contains the breadcrumbs which will be sent with the event
+ */
+@property(nonatomic, strong) NSMutableArray<SentryBreadcrumb *> *breadcrumbArray;
+
+/**
+ * The release version name of the application.
+ */
+@property(atomic, copy) NSString *_Nullable releaseString;
+
+/**
+ * This distribution of the application.
+ */
+@property(atomic, copy) NSString *_Nullable distString;
+
+/**
+ * The environment used in this scope.
+ */
+@property(atomic, copy) NSString *_Nullable environmentString;
+
+/**
+ * Set the fingerprint of an event to determine the grouping
+ */
+@property(atomic, strong) NSArray<NSString *> *_Nullable fingerprintArray;
+
+/**
+ * SentryLevel of the event
+ */
+@property(atomic) enum SentryLevel levelEnum;
+
+@end
+
+@implementation SentryScope
+
+#pragma mark Initializer
+
+- (instancetype)init {
+    if (self = [super init]) {
+        self.listeners = [NSMutableArray new];
+        [self clear];
+    }
+    return self;
+}
+
+- (instancetype)initWithScope:(SentryScope *)scope {
+    if (self = [super init]) {
+        self.listeners = [NSMutableArray new];
+        self.extraDictionary = scope.extraDictionary.mutableCopy;
+        self.tagDictionary = scope.tagDictionary.mutableCopy;
+        SentryUser *scopeUser = scope.userObject;
+        SentryUser *user = nil;
+        if (nil != scopeUser) {
+            user = [[SentryUser alloc] init];
+            user.userId = scopeUser.userId;
+            user.data = scopeUser.data.mutableCopy;
+            user.username = scopeUser.username;
+            user.email = scopeUser.email;
+        }
+        self.userObject = user;
+        self.contextDictionary = scope.contextDictionary.mutableCopy;
+        self.breadcrumbArray = scope.breadcrumbArray.mutableCopy;
+        self.releaseString = scope.releaseString;
+        self.distString = scope.distString;
+        self.environmentString = scope.environmentString;
+        self.levelEnum = scope.levelEnum;
+        self.fingerprintArray = scope.fingerprintArray.mutableCopy;
+    }
+    return self;
+}
+
+#pragma mark Global properties
+
+- (void)addBreadcrumb:(SentryBreadcrumb *)crumb {
+    [SentryLog logWithMessage:[NSString stringWithFormat:@"Add breadcrumb: %@", crumb] andLevel:kSentryLogLevelDebug];
+    @synchronized (self) {
+        [self.breadcrumbArray addObject:crumb];
```

Comment:
Now, all breadcrumbs are stored indefinitely into RAM causing a memory leak.

Reply:
Yes, this sadly got merged in here: https://github.com/getsentry/sentry-cocoa/pull/348/files#diff-79099d807271a5ec576d48da75162a5fL50

We'll get this fixed, thanks @tmm1 for spotting this.

Reply:
We merged a fix and will be releasing 5.0.0-beta.5 soon:

https://github.com/getsentry/sentry-cocoa/pull/450